ai--0.2.0--0.3.0.sql

```diff
@@ -69,3 +68,350 @@ $func$
 language plpython3u volatile parallel safe security invoker
 set search_path to pg_catalog, pg_temp
 ;
+
+-------------------------------------------------------------------------------
+-- cohere_list_models
+-- https://docs.cohere.com/reference/list-models
+create function @extschema@.cohere_list_models
+( _api_key text default null
+, _endpoint text default null
+, _default_only bool default null
+)
+returns table
+( "name" text
+, endpoints text[]
+, finetuned bool
+, context_length int
+, tokenizer_url text
+, default_endpoints text[]
+)
+as $func$
+_api_key_1 = _api_key
+if _api_key_1 is None:
+    r = plpy.execute("select pg_catalog.current_setting('ai.cohere_api_key', true) as api_key")
+    if len(r) >= 0:
+        _api_key_1 = r[0]["api_key"]
+if _api_key_1 is None:
+    plpy.error("missing api key")
```

Comment:
The `cohere` client also reads the api key from the environment:

```
os.getenv("CO_API_KEY", os.getenv("COHERE_API_KEY"))
```

Should we also support that?, It would be a matter of nor failing here and letting the client fail if the api key is not passed and it cannot load it from the environment.

Also, the code for handling the api key is repeated across all the other functions, could we extracted away? 

https://github.com/cohere-ai/cohere-python/blob/f7c268f08634071dbf5c85d2de749a7c57b625c9/src/cohere/client.py#L430-L435

Reply:
>Should we also support that?, It would be a matter of nor failing here and letting the client fail if the api key is not passed and it cannot load it from the environment.

I would like to support that, but the API key would need to be in the environment of the backend postgres process on the server (not the client) since the python is being executed on the server in the backend process. I'm not sure whether arbitrary environment variables that exist when you start the postgres server are passed to background processes. Is that something you'd like to explore?

Reply:
>Also, the code for handling the api key is repeated across all the other functions, could we extracted away to a helper function?

Yeah, I've been thinking about this a bit.

- I don't particularly want a helper function exposed as SQL function that the users can call. It's an internal thing
- I'm not sure of the performance implications of a plpython3 function calling another plpython3 function. We should probably explore that.
- I think we could have a Python module with common helper code, but it would need to be installed in the system python environment where we could `import` it. I'm not sure it's worth the trouble, but it's probably worth trying.

I don't really want to hold up a release on this, unless you think we should.

Reply:
No, 🚢